pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4219,6 +4221,36 @@ int uprobe_seq_ops_trigger(struct pt_regs *ctx)
     return 0;
 }
 
+SEC("uprobe/trigger_mem_dump_event")
+int uprobe_mem_dump_trigger(struct pt_regs *ctx)
+{
+    u64 address = 0;
+    u32 trigger_pid = bpf_get_current_pid_tgid() >> 32;
+
+#if defined(bpf_target_x86)
+    address = ctx->bx; // 1nd arg
+    char arch[] = "x86_64";
+#elif defined(bpf_target_arm64)
+    address = ctx->user_regs.regs[1]; // 1nd arg
+    char arch[] = "arm64";
+#else
+    return 0;
+#endif
+
+    event_data_t data = {};
+    if (!init_event_data(&data, ctx))
+        return 0;
+
+    // uprobe was triggered from other tracee instance
+    if (data.config->tracee_pid != trigger_pid)
+        return 0;
+
+    save_to_submit_buf(&data, (void *) &address, sizeof(void *), 0);
+    save_bytes_to_buf(&data, (void *) address, MEM_DUMP_SIZE, 1);
```

Comment:
Better if you get MEM_DUMP_SIZE as an argument and AND it with some max allowed size.
Also, since the whole point of the event is to read these bytes and this operation can fail (e.g. memory not mapped), I think you should check for error here and return if failed.